### DISCUSSION TITLE
que lea la cadena con los espacios y leer el largo

### DISCUSSION THREAD

**Discussion by Elias-Turner** [INGCOM-UNRN-P1/cursada-2023#108] [Preguntas y respuestas] [answered]
chicos surgió un problema con la parte que cuanto escribo algo con espacios, la función `largo_seguro` solo toma la primera parte.
Es decir, si yo ingreso: `"hola soy elias"`, me dice que el largo es de 4, pero cuando pongo la forma de leer los espacio incluidos (`"%[^\n]"`) el programa no me deja ingresar el texto y me pone que el largo es 0.
Saben como puedo ingresar una cadena con espacios y que me lo lea??

Segunda pregunta, tiene que leer el largo o la capacidad?. Es decir, si pongo que la capacidad es `7`, pero ingreso un largo de `34`, la función tiene que indicarme `34` o `7`??

**Comment by martinvilu** [comment]
Qué usaste para el scanf?

La capacidad es fija, el largo depende de la cadena.

**Reply by Elias-Turner** [reply]
use: "%s\n", cadena

**Reply by Nikzza** [reply]
eso lo usaste para hacer printf? o para escanear?

**Reply by Elias-Turner** [reply]
para escanear

**Reply by martinvilu** [reply, marked as answer]
Probablemente, lo mejor sea usar `fgets`
```C
#include <stdio.h>
#define CADENA_MAX 256

int main()
{
    char string[CADENA_MAX];
    printf ("Ingresa una cadena:");
    fgets (string);
    printf ("la cadena es '%s'\n", string);
    return 0;
}
```
La cadena cargada con esta función incluye el `\n` además del `\0`.

**Reply by martinvilu** [reply]
Solo una cosa, con `scanf` es suficiente para probar las funciones, el `fgets` es una opcion para cadenas mas largas.

**Comment by Nikzza** [comment]
Respuesta a lo segundo: en ese caso te tendria que indicar overflow porque la capacidad de tu cadena es mas chica que la cantidad de palabras que queres ingresar
